### PR TITLE
Set up CI

### PR DESCRIPTION
**Pull request by infinisil**
Based on #1 

Very simple `nix-build` for the main build, the shell and the Nixpkgs sanity check

### CODE REVIEW

**Review by philiptaron — APPROVED** [review]
Yeah, was about to open an issue for it ;-)

I like the DetSys stuff better but this works just fine.

**Comment by infinisil** [comment]
Weird, should've triggered automatically, investigating

**Comment by philiptaron** [comment]
It's because I didn't click on #1 merge yet.

**Comment by infinisil** [comment]
Ah right!

**Comment by philiptaron** [comment]
Maybe you need to rebase?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: CI
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
```
Comment (infinisil) [inline .github/workflows/main.yml:8]:
```suggestion
on:
  pull_request:
    branches:
      - main
  push:
    branches:
      - main
```

Whoops :)

**Review by philiptaron — APPROVED** [review]
My brain completely yadda-yadda-yadda'd "master" vs. "main" difference. Whoops.

**Pull request merged**